src/braket/default_simulator/openqasm/_helpers/arrays.py

```diff
@@ -41,6 +42,8 @@ def convert_range_def_to_slice(range_def: RangeDefinition) -> slice:
     """Convert AST node into Python slice object"""
     buffer = np.sign(range_def.step.value) if range_def.step is not None else 1
     start = range_def.start.value if range_def.start is not None else None
+    if isinstance(range_def.end, UnaryExpression):
+        raise ValueError("Cannot measure non qubit or register values.")
```

Comment:
This error seems out of place. I'm imagining you were motivated to add this because you got an error in the measurement visitor function. Make sure you're calling `self.visit` on the index before accessing it. I'll leave more detailed comments on that part of the code